src/audio.c

```diff
@@ -1338,17 +1364,15 @@ int audio_encoder_set(struct audio *a, const struct aucodec *ac,
 	if (ac->ptime)
 		tx->ptime = ac->ptime;
 
-	if (!tx->ausrc) {
-		err |= audio_start(a);
-	}
-
```

Comment:
Is this a good idea ?

Reply:
yes, I suggest to define a clear hierarchy
- `audio_update` / `mediatrack_start_audio`
  - `aufilt_setup`
  - `audio_decoder_set`
  - `audio_encoder_set`
  - `start_source`

And each of this functions behave according to their name

Reply:
For the decoding stream it is like that:
- `audio_update` / `media_start_audio`
  - `aufilt_setup`
  - `audio_decoder_set`
    - `aurecv_start_player`

Because the decoder can be changed during a call. This makes it sometimes necessary to restart the audio player when `audio_decoder_set` is called.